airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -292,18 +302,24 @@ def list_keys(
             'PageSize': page_size,
             'MaxItems': max_items,
         }
-
         paginator = self.get_conn().get_paginator('list_objects_v2')
         response = paginator.paginate(
-            Bucket=bucket_name, Prefix=prefix, Delimiter=delimiter, PaginationConfig=config
+            Bucket=bucket_name,
+            Prefix=prefix,
+            Delimiter=delimiter,
+            PaginationConfig=config,
+            StartAfter=start_after_key,
+        )
+        # JMESPath to query directly on paginated results
+        filtered_response = response.search(
+            "Contents[?to_string("
+            "LastModified)<='\"{}\"' && "
+            "to_string(LastModified)>='\"{"
```

Comment:
if it's going to be called `start_after_datetime` then this should be strictly greater than, i think

Reply:
but i recommend just `from_datetime` and `to_datetime`

Reply:
sure. `from_datetime` and `to_datetime` make more sense. I will do the change.

Reply:
I have removed `from_datetime`, `to_datetime` and using `ResponseFilter` class with the filter method which supports all the operations defined by user: [commit](https://github.com/apache/airflow/pull/16821/commits/4ef750ea486f76c98f7e5927b12fc62b4767913c)